src/components/tabs/Tabs.js

```diff
@@ -3,7 +3,7 @@ import { TabContent, Tab, Tablink, TabNav } from './';
 export class Tabs extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { id: 0 };
+    this.state = { id: this.props.selectedIndex || 0 };
```

Comment:
you should set  `selectedIndex` via componentWillReceiveProps since it is in props.